s3torchbenchmarking/src/s3torchbenchmarking/dcp/config.yaml

```diff
@@ -0,0 +1,16 @@
+s3:
+  region: ???
+  uri: ???
+epochs: 4
+path: nvme # only used when `checkpoint.storage` contains `disk`, ignored for `s3`
```

Comment:
Can we reuse configs from `conf/checkpoint/*` to define configuration for disk and s3?

Reply:
Additionally, can we move this config to `s3torchbenchmarking/conf` directory, that all configs were located in one place.

Reply:
My overall approach for this CR was to bundle everything under `dcp`, so it lives on its own folder/package; will move things around